HIP/hip-906.md

```diff
@@ -95,13 +93,42 @@ interface IHRC632 {
 }
 ```
 
-Once the above functionality has been implemented in services, an EOA or contract with hbars will be able to call the `hbarAllowance` and `hbarApprove` functions as follows:
+Once the above functionality has been implemented in services, an EOA with hbars will be able to call the `hbarAllowance` and `hbarApprove` functions as follows:
 
 ```
 IHRC(accountAddress).hbarAllowance(address spender)
 IHRC(accountAddress).hbarApprove(address spender, uint256 amount)
 ```
-> The `hbarApprove` function will be callable by either an EOA or a contract as long as the signatures required by the security model are met.
+> The `hbarApprove` function will be callable by an EOA as long as the signatures required by the security model are met.
+
+## Acting on a Contract
+The `IHederaAccountService` system contract will be updated to include the `hbarAllowance` and `hbarApprove` functions.  
+The signature of these calls will be similar to the EOA calls except that they will include the contract address as the first argument.
+
+The following table describes the function selector for the new `hbarAllownace` and `hbarApprove` functions and the associated function signature and response in `IHederaAccountService`.
+
+| Function Selector Hash | Function Signature                                          | Response                                                                               | 
+|------------------------|-------------------------------------------------------------|----------------------------------------------------------------------------------------|
+| 0xfec46666             | hbarAllowance(address owner, address spender)               | (ResponseCode, uint256 - amount of hbar allowances currently available to the spender) | 
+| 0x6e0c21cc             | hbarApprove(address owner, address spender, uint256 amount) | ResponseCode                                                                           |
+
+The solidity interface for `IHederaAccountService` will be the following
+
+```
+interface IHederaAccountService {
+    function hbarAllowance(address owner, address spender) external returns (responseCode, int256);
```

Comment:
Shouldn't there be an ellipsis here because there are already methods in `IHederaAccountService`?  Actually the wording at L115 could indicate these are _additions_ too. 

Or maybe provide here, for clarity, the entire `IHederaAccountService` as it appears after this addition?  (Though it will be in the smart contract repo, so probably not necessary.)